### PR TITLE
Add poetry version used to generate lock file

### PR DESCRIPTION
**Pull request by pfouque**
# Pull Request Check List

Resolves: https://github.com/python-poetry/poetry/pull/2773#issuecomment-1353657313

Like in this comment, we would like to ensure the committed lock file has been generated with a specific version.

Following #2773 This PR simply add the poetry version used to generate the lock file (with the same fingerprint than `poetry --version`)

From
```
# This file is automatically @generated by Poetry and should not be changed by hand.
```
To
```
# This file is automatically @generated by Poetry (version 1.3.1) and should not be changed by hand.
```

- [ ] Added **tests** for changed code.
- [ ] Updated **documentation** for changed code.

### CODE REVIEW

**Comment by radoering** [comment]
It might make more sense to add it in the metadata section so it can be read more easily by other tools?

**Comment by radoering** [comment]
Thinking about it again, the advantage of the comment is that the lock file is not rewritten if there are no changes in the actual content. So there are no unnecessary diffs.

**Comment by pfouque** [comment]
@radoering metadata is a great idea! 
There is already `python-version` and `lock-version`, adding `poetry-version` in-between would make a lot of sense...

**Comment by neersighted** [comment]
I am a big -1 on `poetry-version` -- the reason being that there are two uses for this information. The first is diagnostic/informational; a comment here is better to avoid conflicts and unnecessary churn, as well as to discourage the second use.

The second category of users want to infer things about the contents of the lock file based on the Poetry version. That is in correct, and is in fact what the lock file version aims to solve. The solution here is to start writing down documentation on the lock file, which also helps explain when things change despite no bump (usually due to bug fixes in Poetry).

File: src/poetry/packages/locker.py [hunk 1]
```diff
@@ -40,8 +41,8 @@
 logger = logging.getLogger(__name__)
 _GENERATED_IDENTIFIER = "@" + "generated"
 GENERATED_COMMENT = (
-    f"This file is automatically {_GENERATED_IDENTIFIER} by Poetry and should not be"
-    " changed by hand."
+    f"This file is automatically {_GENERATED_IDENTIFIER} by Poetry (version"
+    f" {__version__}) and should not be changed by hand."
```
Comment (radoering) [inline src/poetry/packages/locker.py:45]:
```suggestion
    f"This file is automatically {_GENERATED_IDENTIFIER} by Poetry"
    f" {__version__} and should not be changed by hand."
```

Should be sufficient and is consistent with what we write into the INSTALLER file in #6205.

**Comment by pfouque** [comment]
@neersighted After some tests, I understand your point about "lock-version" and I agree it doesn't bring much value in metadata.
Thanks @radoering, I do not have a strong opinion on the exact format, my proposition was based on the output of "poetry --version", but whatever the format is, the aim is to give access to this information for diagnostic.

**Review by radoering — APPROVED** [review]

**Comment by pfouque** [comment]
🙏 Thanks!

**Pull request merged**